### ISSUE TITLE
numpy `VisibleDeprecationWarning` when converting lists to numpy array in specific context

### ISSUE THREAD

**Issue by falkben** [ExoCTK/exoctk#466]
On the line of code below, we are converting a list of lists to a numpy array. This emits the following numpy deprecation warning:

> VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.

```python
grouped_badPAs = np.asarray(grouped_badPAs)
```

[Code](https://github.com/ExoCTK/exoctk/blob/develop/exoctk/contam_visibility/visibilityPA.py#L425)

If we switch to using the following instead, as suggested by the warning, it will remove the warning:

```python
np.asarray(grouped_badPAs, dtype=object)
```

And the output appears to be the same, though this requires further testing.

This was deprecated in ver. 1.19 in Numpy: [link](https://numpy.org/devdocs/release/1.19.0-notes.html?highlight=ragged#deprecate-automatic-dtype-object-for-ragged-input)

This version of numpy is specified in the environment files (e.g.: [environment-3.7.yml](https://github.com/ExoCTK/exoctk/blob/master/env/environment-3.7.yml)).

More information here: https://numpy.org/neps/nep-0034-infer-dtype-is-object.html

The reason this is an issue for us is that our continuous integration pipeline uses `coverage`, which calls into this exoctk function, and coverage seems to make these warnings exceptions, and so fails our builds. 🙃

**Comment by hover2pi** [comment]
Seems like an easy fix @falkben ! Thanks for making the Issue. We'll make sure to include it in the `contam_visibility` optimization PR.

**Comment by hover2pi** [comment]
Fixed in #492
